packages/serialize/src/lib/Serialize.ts

```diff
@@ -90,62 +104,80 @@ export namespace Serialize {
   }
 
   function fromJSONArray(json: JSON[]): unknown[] {
-    return json.reduce<unknown[]>((raw, value) => [...raw, fromJSON(value)], []);
+    const arr: unknown[] = [];
+
+    for (const value of json) arr.push(fromJSON(value));
+    return arr;
   }
 
   function fromJSONMap(json: [PropertyKey, JSON][]): [PropertyKey, unknown][] {
-    return json.reduce<[PropertyKey, unknown][]>((raw, [key, value]) => [...raw, [key, fromJSON(value)]], []);
+    const arr: [PropertyKey, unknown][] = [];
+
+    for (const [key, value] of json) arr.push([key, fromJSON(value)]);
+    return arr;
   }
 
   function fromJSONObject(json: Record<PropertyKey, JSON>): Record<PropertyKey, unknown> {
-    return Object.entries(json).reduce<Record<PropertyKey, unknown>>((raw, [key, value]) => ({ ...raw, [key]: fromJSON(value) }), {});
+    const obj: Record<PropertyKey, unknown> = {};
+
+    for (const key in json) {
+      if ({}.hasOwnProperty.call(json, key)) {
+        obj[key] = fromJSON(json[key]);
+      }
+    }
+
+    return obj;
   }
 
+  export enum Keying {
+    Type = 't',
+    Value = 'v'
+  }
   /**
    * The json format type interface.
    * @since 1.0.0
    */
   export interface JSON {
     /**
-     * The type of {@link JSON.value}
+     * The type of {@link JSON.v}
      * @since 1.0.0
      */
-    type: Type;
+    [Keying.Type]: Type;
 
     /**
      * The value for this json.
      * @since 1.0.0
      */
-    value: string | number | boolean | null | JSON[] | [PropertyKey, JSON][] | Record<PropertyKey, JSON> | { source: string; flags: string };
+    [Keying.Value]: string | number | boolean | null | JSON[] | [PropertyKey, JSON][] | Record<PropertyKey, JSON> | { s: string; f: string };
```

Comment:
Shouldn't we make the regex keys into enums too? I think we should or just remove the enums altogether.